### ISSUE TITLE
REPL buffer scrolls uncontrollably

### ISSUE THREAD

**Issue by arichiardi** [clojure-emacs/inf-clojure#121]
This is quite annoying in my opinion, every time the point is on a symbol (say `def` in `lumo`) that does not have arglists and then one that has arglists, you see the REPL buffer scrolling up or down.

I dug into it and it seems that [this var](https://github.com/emacs-mirror/emacs/blob/master/lisp/comint.el#L402) in comint sets `comint-postoutput-scroll-to-bottom` which is responsible for the scrolling.

I could not find a way to customize the behavior, except from removing it from the var completely with:

```elisp
(seq-filter (lambda (sym) (not (eq 'comint-postoutput-scroll-to-bottom sym))) comint-output-filter-functions)
```

I know it is extreme but I would like to set this as default. An alternative would be to pass in our own `comint-output-filter-functions` that checks what is the REPL return and decides whether to scroll or not - it seems quite some work for no gain.

Thoughts?

**Comment by bbatsov** [comment]
> This is quite annoying in my opinion, every time the point is on a symbol (say def in lumo) that does not have arglists and then one that has arglists, you see the REPL buffer scrolling up or down.

Can you share some gif recording of this or something? I'm having a hard time understanding what exactly is happening.

**Comment by arichiardi** [comment]
Yep I will, you can try to open an `inf-clojure` project and move point from `def` to any other symbol. You will see the REPL buffer scrolling.

**Comment by bbatsov** [comment]
I guess what puzzles me is what is this output that's triggering this.

**Comment by arichiardi** [comment]
Looks like comint calculates the amount it scrools based on the characters returned in the response, but we hide the returned characters for our actions. So it scrolls but really should not.

**Comment by arichiardi** [comment]
What should we do about this? defcustom?

**Comment by MalloZup** [comment]
autogenerated with https://github.com/MalloZup/doghub:  issue inactive since 450 days. Please update the issue or close it
